cache/lib/cache/s3.ex

```diff
@@ -46,4 +46,101 @@ defmodule Cache.S3 do
       {:error, _reason} -> false
     end
   end
+
+  @doc """
+  Uploads an artifact to S3.
+
+  Returns :ok on success or {:error, reason} on failure.
+  If the local file does not exist, returns :ok (the file may have been evicted).
+  """
+  def upload(account_handle, project_handle, artifact_id) do
+    key = Cache.Disk.cas_key(account_handle, project_handle, artifact_id)
+    local_path = Cache.Disk.artifact_path(key)
+
+    Logger.info("Starting S3 upload for artifact: #{key}")
+
+    case upload_file(key, local_path) do
+      :ok ->
+        Logger.info("Successfully uploaded artifact to S3: #{key}")
+        :ok
+
+      {:error, reason} ->
+        Appsignal.send_error(%RuntimeError{message: "Failed to upload artifact to S3"}, %{
+          key: key,
+          reason: reason
+        })
+
+        {:error, reason}
+    end
```

Comment:
```suggestion
    :ok = upload_file(key, local_path) do
    Logger.info("Successfully uploaded artifact to S3: #{key}")
```

I think we discussed it somewhere else – any reason why not let Appsignal just pick up the crash instead of us invoking that manually since you'll get the stacktrace anyways?